library/catalog/views.py

```diff
@@ -181,13 +183,26 @@ def change_review_status(request, book_id, review_id, new_status):
 
 
 def home(request):
-    top_books = Book.objects.filter(status=Status.ACTIVE).order_by('-statistic__rating')[:20]
-    new_books = Book.objects.filter(status=Status.ACTIVE).order_by('-pk')[:20]
-    books_genres = []
-    for genres_list in Book.objects.values('genres'):
-        for genre in genres_list.get('genres'):
-            if genre and genre not in books_genres:
-                books_genres.append(genre)
+    if cache_storage.get_cache('books_rating_cache') is None:
```

Comment:
top_books = cache_storage.get_value('top_books ')
if top_books is None:
    sorted_books = sorted(Book.objects.filter(status=Status.ACTIVE), key=lambda book: book.statistic.rating,
                          reverse=True)[:20]
    cache_storage.add_cache('books_rating_cache', top_books)